python/hsfs/core/feature_group_engine.py

```diff
@@ -314,17 +316,78 @@ def insert_stream(
 
         return streaming_query
 
-    def _save_feature_group_metadata(
-        self, feature_group, feature_dataframe, write_options
+    def _check_and_save_feature_group_metadata(
```

Comment:
I understand what you mean. I went forth and back between your suggestion and having them in on function a couple of times . I don't mind going back another time. 

I wanted to make the code a little bit more understandable for outsiders with `feature_group_not_exists ` instead of `not feature_group.id`, because vice versa it took me some time to understand what `not feature_group.id` entails. But guess a comment would also do it justice.